smt.go

```diff
@@ -438,8 +503,16 @@ func (tree *BASSparseMerkleTree) Commit(recentVersion *Version) (Version, error)
 	if recentVersion != nil {
 		tree.recentVersion = *recentVersion
 	}
+
+	currentSize, releasableSize := tree.root.size(tree.recentVersion)
+	if releaseVersion := tree.gcStatus.pop(currentSize); releaseVersion > 0 {
+		size := tree.root.release(releaseVersion)
```

Comment:
The tree gc can only release child of root, and the granularity is a little bit large.